### PR TITLE
Filtering irrelevant media during enrichment

### PR DESCRIPTION
**Pull request by jamesbraza**
This PR adds filtration of irrelevant media using the enrichment step. Enrichment labels media as relevant or irrelevant, and we use this to in-place remove the media.

For example, with the Wikimedia logos in `tests/stub_data/duplicate_media.pdf`, the enrichment is:

```none
**IRRELEVANT:** The media is a logo for Wikimedia. It does not contain scientific content such as figures, charts, tables, equations, or data visualizations that could help answer scientific questions.
```

---

> [!NOTE]
> Enrichment now labels media as RELEVANT/IRRELEVANT, parses the label to set is_irrelevant, filters irrelevant media in-place, and updates prompts and tests.
> 
> - **Enrichment & Filtering**:
>   - `Settings.make_media_enricher`: parse LLM enrichment label to set `media.info["is_irrelevant"]` and `"enriched_description"`, then remove irrelevant media in-place from `ParsedText.content`.
>   - Enrichment summary now includes `filtered=` count alongside `enriched=` and `radius=`.
> - **Prompts**:
>   - Update `individual_media_enrichment_prompt_template` and `full_page_enrichment_prompt_template` to require responses start with `RELEVANT:` or `IRRELEVANT:` and then the description.
> - **Utils**:
>   - Add `parse_enrichment_irrelevance(enrichment: str) -> tuple[bool, str]` to extract relevance label and strip it from the description.
> - **Tests**:
>   - Add `test_media_enrichment_filters_irrelevant` validating label parsing, in-place filtration, and summary counts.
>   - Minor test import update to include `Settings`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a93f0202b6f9e54a082042436e7a171585703fbc. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/paper-qa-nemotron/tests/test_paperqa_nemotron.py [hunk 1]
```diff
@@ -374,3 +374,66 @@ def test_pad_image_with_border(subtests: pytest.Subtests) -> None:
...
+    assert filtered_count > 0, "Expected some filtration to take place"
+
+    # Game the in-place change of enrichment to get Wikimedia logos
+    wikimedia_media_after = [
+        m
+        for m in media_before
+        if isinstance(m.info["enriched_description"], str)
+        and "wikimedia" in m.info["enriched_description"].lower()
+    ]
+    assert len(wikimedia_media_after) <= 1, "Expected most Wikimedia logos to be gone"
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply packages/paper-qa-nemotron/tests/test_paperqa_nemotron.py]: Oh nice catch, fixed this just now

File: packages/paper-qa-nemotron/tests/test_paperqa_nemotron.py [hunk 2]
```diff
@@ -374,3 +374,66 @@ def test_pad_image_with_border(subtests: pytest.Subtests) -> None:
...
+        assert media.info["enriched_description"], "Expected enriched description"
+    filtered_count = int(
+        enrichment_summary.split("filtered=", maxsplit=1)[1].split("|")[0]
+    )
+    assert (
+        len(media_before) - len(media_after) == filtered_count
+    ), "Filtered summary mismatches actual filtration"
+    assert filtered_count > 0, "Expected some filtration to take place"
+
+    # Game the in-place change of enrichment to get Wikimedia logos
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply packages/paper-qa-nemotron/tests/test_paperqa_nemotron.py]: Actually this comment is what I meant, but I clarified it just now

File: src/paperqa/utils.py [hunk 3]
```diff
@@ -642,3 +642,27 @@ def clean_possessives(text: str) -> str:
...
+    # Handle optional Markdown formatting (e.g. **RELEVANT:** or *IRRELEVANT:*)
+    normalized_start = enrichment.upper().lstrip("*").lstrip()
+    if normalized_start.startswith("IRRELEVANT:"):
+        label_match = re.match(
+            r"^\*{0,2}IRRELEVANT:\*{0,2}\s*", enrichment, re.IGNORECASE
+        )
+        is_irrelevant: bool = True
+    elif normalized_start.startswith("RELEVANT:"):
+        label_match = re.match(
+            r"^\*{0,2}RELEVANT:\*{0,2}\s*", enrichment, re.IGNORECASE
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply src/paperqa/utils.py]: Oh sure, went with this locally just now

File: src/paperqa/utils.py [hunk 4]
```diff
@@ -642,3 +642,27 @@ def clean_possessives(text: str) -> str:
...
+        )
+        is_irrelevant = False
+    else:  # No explicit label matched, conservatively assume relevant
+        return False, enrichment.strip()
+    no_label_enrichment = enrichment[label_match.end() :] if label_match else enrichment
+    return is_irrelevant, no_label_enrichment.strip()
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply src/paperqa/utils.py:664]: It's covered as part of the new `test_media_enrichment_filters_irrelevant`

[the remark this answers is not in the document]
Comment (jamesbraza) [reply src/paperqa/utils.py:664]: I agree a unit test covering edge cases would be nice, but this code is semi-hacky, so ima YAGNI on more testing

File: packages/paper-qa-nemotron/tests/test_paperqa_nemotron.py [hunk 5]
```diff
@@ -374,3 +374,66 @@ def test_pad_image_with_border(subtests: pytest.Subtests) -> None:
...
+    wikimedia_media_after = [
+        m
+        for m in media_before
+        if isinstance(m.info["enriched_description"], str)
+        and "wikimedia" in m.info["enriched_description"].lower()
+    ]
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply packages/paper-qa-nemotron/tests/test_paperqa_nemotron.py:438]: Oh nice catch, fixed this just now

File: src/paperqa/utils.py [hunk 6]
```diff
@@ -642,3 +642,27 @@ def clean_possessives(text: str) -> str:
...
+        )
+        is_irrelevant: bool = True
+    elif normalized_start.startswith("RELEVANT:"):
+        label_match = re.match(
+            r"^\*{0,2}RELEVANT:\*{0,2}\s*", enrichment, re.IGNORECASE
+        )
+        is_irrelevant = False
+    else:  # No explicit label matched, conservatively assume relevant
+        return False, enrichment.strip()
+    no_label_enrichment = enrichment[label_match.end() :] if label_match else enrichment
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply src/paperqa/utils.py:663]: I aligned them locally just now, thanks
